apollo-client/src/main/java/com/ctrip/framework/apollo/monitor/internal/ApolloClientMonitorContext.java

```diff
@@ -0,0 +1,96 @@
+
+import com.ctrip.framework.apollo.monitor.api.ApolloClientBootstrapArgsMonitorApi;
+import com.ctrip.framework.apollo.monitor.api.ApolloClientExceptionMonitorApi;
+import com.ctrip.framework.apollo.monitor.api.ApolloClientNamespaceMonitorApi;
+import com.ctrip.framework.apollo.monitor.api.ApolloClientThreadPoolMonitorApi;
+import com.ctrip.framework.apollo.monitor.internal.exporter.ApolloClientMetricsExporter;
+import com.ctrip.framework.apollo.monitor.internal.exporter.impl.NullApolloClientMetricsExporter;
+import com.ctrip.framework.apollo.monitor.internal.listener.ApolloClientMonitorEventListener;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.NullClientBootstrapArgsMonitorApi;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.NullClientExceptionMonitorApi;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.NullClientNamespaceMonitorApi;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.NullClientThreadPoolMonitorApi;
+import com.google.common.collect.Lists;
+import java.util.List;
+
+/**
+ * @author Rawven
+ */
+public class ApolloClientMonitorContext {
+
+  private ApolloClientExceptionMonitorApi apolloClientExceptionMonitorApi = new NullClientExceptionMonitorApi();
+  private ApolloClientNamespaceMonitorApi apolloClientNamespaceMonitorApi = new NullClientNamespaceMonitorApi();
+  private ApolloClientBootstrapArgsMonitorApi apolloClientBootstrapArgsMonitorApi = new NullClientBootstrapArgsMonitorApi();
+  private ApolloClientThreadPoolMonitorApi apolloClientThreadPoolMonitorApi = new NullClientThreadPoolMonitorApi();
+  private ApolloClientMetricsExporter apolloClientMetricsExporter = new NullApolloClientMetricsExporter();
+
+  public void setApolloClientExceptionMonitorApi(
+      ApolloClientExceptionMonitorApi apolloClientExceptionMonitorApi) {
+    this.apolloClientExceptionMonitorApi = apolloClientExceptionMonitorApi;
+  }
+
+  public void setApolloClientNamespaceMonitorApi(
+      ApolloClientNamespaceMonitorApi apolloClientNamespaceMonitorApi) {
+    this.apolloClientNamespaceMonitorApi = apolloClientNamespaceMonitorApi;
+  }
+
+  public void setApolloClientBootstrapArgsMonitorApi(
+      ApolloClientBootstrapArgsMonitorApi apolloClientBootstrapArgsMonitorApi) {
+    this.apolloClientBootstrapArgsMonitorApi = apolloClientBootstrapArgsMonitorApi;
+  }
+
+  public void setApolloClientThreadPoolMonitorApi(
+      ApolloClientThreadPoolMonitorApi apolloClientThreadPoolMonitorApi) {
+    this.apolloClientThreadPoolMonitorApi = apolloClientThreadPoolMonitorApi;
+  }
+
+  public void setApolloClientMetricsExporter(
+      ApolloClientMetricsExporter apolloClientMetricsExporter) {
+    this.apolloClientMetricsExporter = apolloClientMetricsExporter;
+  }
+
+  public List<ApolloClientMonitorEventListener> getCollectors() {
+    return Lists.newArrayList(
+        (ApolloClientMonitorEventListener) apolloClientBootstrapArgsMonitorApi,
```

Comment:
That will cause class cast exception if use Null impl.

can we add a method to prepare listeners in memory?

1. method `addListener`
2. method `setListenerList`

and method `getListenerList` to get them directly.